lib/vector-config/src/http.rs

```diff
@@ -0,0 +1,35 @@
+use http::StatusCode;
+use serde_json::Value;
+use std::cell::RefCell;
+
+use crate::{
+    schema::{generate_number_schema, SchemaGenerator, SchemaObject},
+    Configurable, GenerateError, Metadata, ToValue,
+};
+
+impl ToValue for StatusCode {
+    fn to_value(&self) -> Value {
+        serde_json::to_value(self.as_u16()).expect("Could not convert status code to JSON")
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏

```suggestion
        serde_json::to_value(self.as_u16()).expect("Could not convert HTTP status code to JSON")
```